Ensure correct ownership for singularity.conf on install

--- a/etc/Makefile.am
+++ b/etc/Makefile.am
@@ -7,3 +7,8 @@ dist_completion_DATA =  bash_completion.d/singularity
 
 
 MAINTAINERCLEANFILES = Makefile.in
+
+# On a Debian system, for instance, we can end up with root.staff ownership
+# after sudo make install
+install-data-hook:
+	[ $$(whoami) = root ] && chown root.root $(sysconfdir)/singularity/singularity.conf || true
