--- singularity-2.0/src/loop-control.c~	2016-06-01 15:12:38.000000000 +0100
+++ singularity-2.0/src/loop-control.c	2016-06-24 12:22:46.120381475 +0100
@@ -120,7 +120,7 @@
     int loop_fd = fileno(loop_fp);
 
     if ( ioctl(loop_fd, LOOP_CLR_FD, 0) != 0 ) {
-        fprintf(stderr, "ERROR: Could not clear loop device\n");
+        fprintf(stderr, "ERROR: Could not clear loop device: %s\n", strerror(errno));
         return(-1);
     }
 

