Fix configured container_dir

--- singularity-2.2.1/etc/singularity.conf~	2017-02-14 15:21:36.000000000 +0000
+++ singularity-2.2.1/etc/singularity.conf	2017-05-21 14:57:49.750949186 +0000
@@ -125,11 +125,11 @@
 
 
 # CONTAINER DIR: [STRING]
-# DEFAULT: /var/singularity/mnt
+# DEFAULT: /var/lib/singularity/mnt
 # This path specifies the location to use for mounting the container, overlays
 # and other necessary file systems for the container. Note, this location
 # absolutely must be local on this host.
-container dir = /var/singularity/mnt
+container dir = /var/lib/singularity/mnt
 
 
 # SESSIONDIR PREFIX: [STRING]
@@ -137,7 +137,7 @@
 # This specifies the prefix for the session directory. Appended to this string
 # is an identification string unique to each user and container. Note, this
 # location absolutely must be local on this host. If the default location of
-# /tmp/ does not work for your system, /var/singularity/sessions maybe a
+# /tmp/ does not work for your system, /var/lib/singularity/sessions maybe a
 # better option.
-#sessiondir prefix = /var/singularity/sessions/
+#sessiondir prefix = /var/lib/singularity/sessions/
 

Diff finished.  Sun May 21 14:59:41 2017
